tests/integration_tests/charts/data/api_tests.py

```diff
@@ -764,6 +764,47 @@ def test_with_virtual_table_with_colons_as_datasource(self):
         assert "':xyz:qwerty'" in result["query"]
         assert "':qwerty:'" in result["query"]
 
+    @pytest.mark.usefixtures("load_birth_names_dashboard_with_slices")
+    def test_with_table_columns_without_metrics(self):
```

Comment:
These tests don't actually rely on the rest of API so personally, suggestion moving these to the `query_context_test.py` and following this [pattern](https://github.com/apache/superset/blob/aef6217a80d8b7d30120efe6502505edc01eb8d3/tests/integration_tests/query_context_tests.py#L849). This is not required, and you can leave it for later if you are busy.

Reply:
will add those integration tests later.